open-metadata-conformance-suite/open-metadata-conformance-suite-server/src/main/java/org/odpi/openmetadata/conformance/tests/repository/instances/TestSupportedRelationshipSharingPropertySearch.java

```diff
@@ -194,28 +197,24 @@ protected void run() throws Exception
                 RepositoryConformanceProfileRequirement.CONSISTENT_TYPES.getProfileId(),
                 RepositoryConformanceProfileRequirement.CONSISTENT_TYPES.getRequirementId());
 
-        String endOneEntityDefGUID = relationshipDef.getEndDef1().getEntityType().getGUID();
-        String endTwoEntityDefGUID = relationshipDef.getEndDef2().getEntityType().getGUID();
-        EntityDef endOneEntityDef = (EntityDef) metadataCollection.getTypeDefByGUID(workPad.getLocalServerUserId(), endOneEntityDefGUID);
-        EntityDef endTwoEntityDef = (EntityDef) metadataCollection.getTypeDefByGUID(workPad.getLocalServerUserId(), endTwoEntityDefGUID);
-
-
-        EntityDef knownEnd1EntityDef = (EntityDef) repositoryHelper.getTypeDefByName(workPad.getLocalServerUserId(), endOneEntityDef.getName());
-        verifyCondition((endOneEntityDef.equals(knownEnd1EntityDef)),
-                assertion0,
-                testTypeName + assertionMsg0,
-                RepositoryConformanceProfileRequirement.CONSISTENT_TYPES.getProfileId(),
-                RepositoryConformanceProfileRequirement.CONSISTENT_TYPES.getRequirementId());
-
-
-        EntityDef knownEnd2EntityDef = (EntityDef) repositoryHelper.getTypeDefByName(workPad.getLocalServerUserId(), endTwoEntityDef.getName());
-        verifyCondition((endTwoEntityDef.equals(knownEnd2EntityDef)),
-                assertion0,
-                testTypeName + assertionMsg0,
-                RepositoryConformanceProfileRequirement.CONSISTENT_TYPES.getProfileId(),
-                RepositoryConformanceProfileRequirement.CONSISTENT_TYPES.getRequirementId());
-
 
+        String end1TypeDefName = relationshipDef.getEndDef1().getEntityType().getName();
+        EntityDef knownEnd1EntityDef = (EntityDef) repositoryHelper.getTypeDefByName(workPad.getLocalServerUserId(), end1TypeDefName);
+        EntityDef end1EntityDef = entityDefs.get(end1TypeDefName);
+        verifyCondition( (end1EntityDef != null) && end1EntityDef.equals(knownEnd1EntityDef),
+                         assertion0,
+                         testTypeName + assertionMsg0,
+                         RepositoryConformanceProfileRequirement.CONSISTENT_TYPES.getProfileId(),
+                         RepositoryConformanceProfileRequirement.CONSISTENT_TYPES.getRequirementId());
+
+        String end2TypeDefName = relationshipDef.getEndDef2().getEntityType().getName();
+        EntityDef knownEnd2EntityDef = (EntityDef) repositoryHelper.getTypeDefByName(workPad.getLocalServerUserId(), end2TypeDefName);
+        EntityDef end2EntityDef = entityDefs.get(end2TypeDefName);
```

Comment:
As above -- if the connector does not implement supertypes, this will return `null` for many relationships and the assertion just below will always fail as a result.

Reply:
Please see additional commit "Refinements to relationship end type checks" which should resolve all the above. Sorry about that.